src/use_case/add_exercise/AddExerciseInteractor.java

```diff
@@ -1,5 +1,34 @@
 package use_case.add_exercise;
 
-public class AddExerciseInteractor {
 
+public class AddExerciseInteractor implements AddExerciseInputBoundary{
+    final AddExerciseDataAccessInterface addExerciseDataAccessObject;
+
+    final AddExerciseOutputBoundary addExercisePresenter;
+
+    public AddExerciseInteractor(AddExerciseDataAccessInterface addExerciseDataAccessObject,
+                                 AddExerciseOutputBoundary addExerciseOutputBoundary) {
+        this.addExerciseDataAccessObject = addExerciseDataAccessObject;
+        this.addExercisePresenter = addExerciseOutputBoundary;
+    }
+
+    @Override
+    public void execute(AddExerciseInputData addExerciseInputData) {
+        int id = addExerciseInputData.getId();
+        String exercise_name = addExerciseInputData.getName();
+
+        // Check routine with id corresponding to id exists
+        if (!addExerciseDataAccessObject.existsById(id)) {
+            //Use exceptions? instead
+            // addExercisePresenter.prepareFailView("Routine" + id + "does not exist"); - temp until presenter implemented
+        // Check exercise with name corresponding to exercise_name exists
+        } else if (!addExerciseDataAccessObject.existsByName(exercise_name)) {
```

Comment:
empty else if? should be calling presenter fail view here (where is fail view getting used?)